### PR TITLE
Add section "Getting started with CAN, CAN FD and CAN XL"

### PR DESCRIPTION
**Pull request by bmenne-dspace**
Provides a simple HowTo to understand how the provided header files can be used by users of the fmi-ls-bus.

### CODE REVIEW

File: ls-bus-guide/4____network_abstraction.adoc [hunk 1]
```diff
@@ -1,43 +1,311 @@
...
+.Summary
+====
+* The LS-BUS API provides macros to write received binary data into a given buffer
+* The buffer is updated by the `FMI3_LS_BUS_BUFFER_WRITE` macro
+* The `FMI3_LS_BUS_BUFFER_WRITE` can be called repeatedly
+====
+
+===== Processing of Bus Operations
+The Bus Operations must now be processed on the receiving side.
+A suitable place for implementation represents `fmi3UpdateDiscreteStates`.
```
[the remark this answers is not in the document]
Comment (bmenne-dspace) [reply ls-bus-guide/4____network_abstraction.adoc:262]: @msuevern I added the requested note. Thanks for pointing that out!

File: ls-bus-guide/4____network_abstraction.adoc [hunk 2]
```diff
@@ -1,43 +1,313 @@
...
+    }
+    ...
+}
+
+----
+<1> Reading the next operation from the `fmi3LsBusUtilBufferInfo` instance.
+<2> Decide which kind of operation needs to be handled.
+<3> Casting into the concrete operation underlying structure.
+
+Please note that using `fmi3UpdateDiscreteStates` requires that the capability flag `providesEvaluateDiscreteStates = "true"` be set within the `modelDescription.xml`.
```
Comment (ClemensB) [inline ls-bus-guide/4____network_abstraction.adoc:300]: The capability flag `providesEvaluateDiscreteStates` should only affect the function `fmi3EvaluateDiscreteStates` (see [2.4.2. Attributes and Capability Flags](https://fmi-standard.org/docs/3.0.1/#common-capability-flags)). `fmi3UpdateDiscreteStates` should always be provided by an FMU supporting the event mode.

File: ls-bus-guide/4____network_abstraction.adoc [hunk 3]
```diff
@@ -1,43 +1,313 @@
...
+`FMI3_LS_BUS_BUFFER_RESET` sets the internal position of the `fmi3LsBusUtilBufferInfo` instance to zero, so that it is essentially emptied and written from the beginning.
+The whole thing is necessary to ensure that Bus Operations that have already been transmitted are not transmitted a second time.
+`FMI3_LS_BUS_CAN_CREATE_OP_CAN_TRANSMIT` now creates a new CAN Transmit operation with the associated parameters such as CAN ID and payload and adds them directly to the `fmi3LsBusUtilBufferInfo` instance.
+Querying the status of a `fmi3LsBusUtilBufferInfo` instance allows you to check whether there is still enough space in the underlying buffer.
+In the last step, `FMI3_LS_BUS_BUFFER_INFO_LENGTH` is used to check whether there are Bus Operations in the respective `fmi3LsBusUtilBufferInfo` variable that should be transmitted in `Event Mode`.
+
+[source,c]
+.Creation of a CAN Transmit operation
+----
+#include "mi3LsBusUtilCan.h"
```
Comment (ClemensB) [inline ls-bus-guide/4____network_abstraction.adoc:102]:
```suggestion
#include "Fmi3LsBusUtilCan.h" 
```

File: ls-bus-guide/4____network_abstraction.adoc [hunk 4]
```diff
@@ -1,43 +1,313 @@
...
+}
+----
+<1> Creation of CAN frame payload.
+<2> Definition of CAN ID of the specified frame.
+<3> Resetting of `fmi3LsBusUtilBufferInfo` variable instance.
+<4> Creation of a CAN Transmit operation and adding it to the specified `fmi3LsBusUtilBufferInfo` variable.
+<5> Verify that free buffer space is available.
+<6> Activate `Event Mode` if needed.
+
+According to exactly the same principle, any specified operation can be created using an existing macro.
```
Comment (ClemensB) [inline ls-bus-guide/4____network_abstraction.adoc:132]:
```suggestion
According to the same principles, any specified operation can be created using the corresponding macro.
```
Der Satz liest sich etwas merkwürdig.

File: ls-bus-guide/4____network_abstraction.adoc [hunk 5]
```diff
@@ -1,43 +1,313 @@
...
+* The CREATE_OP macros are creating a Bus Operation and updating the given buffer in a single step
+====
+
+===== Transmit of Bus Operations
+Within the layered standard, the https://modelica.github.io/fmi-ls-bus/main/#low-cut-variables[connection] between the data to be exchanged (`TX_Data` and `Rx_Data`) and the time of exchange 
+(`Tx_Clock` and `Rx_Clock`) has been well defined.
+The https://modelica.github.io/fmi-ls-bus/main/#low-cut-tx-triggered-clock-variables[simplest case represents] a `triggered` clock as `Tx_Clock` that basically allows to signal events when returning from `fmi3DoStep`.
+
+The program code below schematically illustrates an implementation.
+`fmi3GetClock` is called by the FMU importer after `fmi3DoStep` has completed or ended prematurely.
```
Comment (ClemensB) [inline ls-bus-guide/4____network_abstraction.adoc:147]: I think we should mention here that the importer transitions into event mode using fmi3EnterEventMode first.

File: ls-bus-guide/4____network_abstraction.adoc [hunk 6]
```diff
@@ -1,43 +1,313 @@
...
+Within `fmi3GetClock`, `TX_CLOCK_REFERENCE` represents the valueReference of the respective `Tx_Clock`.
+The usage of the macro `FMI3_LS_BUS_BUFFER_IS_EMPTY` indicates whether there is data to be transferred in the respective buffer.
+If this is the case, the corresponding `Tx_Clock` will tick.
```
Comment (ClemensB) [inline ls-bus-guide/4____network_abstraction.adoc:150]: Note that in theory, the fmi3GetClock may only return true here once per clock activation.

File: ls-bus-guide/4____network_abstraction.adoc [hunk 7]
```diff
@@ -1,43 +1,313 @@
...
+* The LS-BUS C API provides macros to get the START address and LENGTH of the buffer which can be used in the context of `fmi3GetClock` and `fmi3GetBinary`
+====
+
+===== Receive of Bus Operations
+The indication whether new operations are pending within the `Rx_Data` variable is done via the `Rx_Clock`.
+This clock ticks as soon as new data is available.
+The operation-receiving FMU gets the Bus Operations via a `fmi3Binary` variable.
+This variable can be transformed back into a `fmi3LsBusUtilBufferInfo` instance.
+With the help of this transformation, the Bus Operations to be processed can be easily accessed.
+`FMI3_LS_BUS_BUFFER_WRITE` can be used for transformation.
```
Comment (ClemensB) [inline ls-bus-guide/4____network_abstraction.adoc:209]:
```suggestion
The contents of this variable may then be copied into a buffer described by an `fmi3LsBusUtilBufferInfo` instance using `FMI3_LS_BUS_BUFFER_WRITE`.
```

**Pull request merged**